File: L007/digitFreq.java
import java.util.Scanner;

public class digitFreq {
    public static Scanner sc = new Scanner(System.in);

    public static int digitFreq_(long n, int a) {

        int count = 0;

        int temp = (int) n;
        while (temp != 0) {

            int r = temp % 10;

            if (r == a) {

                count++;
            }
            temp = temp / 10;
        }

        return count;

    }

    public static void main(String[] args) {

        int n = sc.nextInt();
        int d = sc.nextInt();

        System.out.println(digitFreq_(n, d));
    }

}

File: L007/digitFreq.cpp
#include<iostream>
using namespace std;


int digitFreq_(long n , int a){


    int count = 0;

    while(n != 0){

        int r = n % 10;

        if( r == a){

            count++;
        }
        n = n/10;

    }

    return count;

}
int main(){
    long n;
    cin>>n;

    int a;
    cin>>a;

    cout<<digitFreq_(n,a);
    return 0;
}